### PR TITLE
NIFI-4651 Add error message and code to PutSQL FlowFile attributes

### PR DESCRIPTION
**Pull request by krisztina-zsihovszki**
# Summary

[NIFI-4651](https://issues.apache.org/jira/browse/NIFI-4651)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 11
  - [x] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by krisztina-zsihovszki** [comment]
recheck

**Comment by Lehel44** [comment]
Thanks for the contribution @krisztina-zsihovszki. Reviewing.

**Review by Lehel44 — CHANGES_REQUESTED** [review]

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutSQL.java [hunk 1]
```diff
@@ -836,7 +870,17 @@ boolean isFragmentedTransactionReady(final List<FlowFile> flowFiles, final Long
...
+    private FlowFile addErrorAttributesToFlowFile(final ProcessSession session, FlowFile flowFile, final Exception exception) {
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(ERROR_MESSAGE_ATTR, exception.getMessage());
 
+        if (exception instanceof SQLException) {
+            attributes.put(ERROR_CODE_ATTR, valueOf(((SQLException) exception).getErrorCode()));
+            attributes.put(ERROR_SQL_STATE_ATTR, valueOf(((SQLException) exception).getSQLState()));
+        }
+
+        return session.putAllAttributes(flowFile, attributes);
```
[the remark this answers is not in the document]
Comment (krisztina-zsihovszki) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutSQL.java:868]: The method adds extra attributes to the FlowFile, it does not remove any existing attribute. (Added a check in unit tests to verify this.)

**Review by Lehel44 — APPROVED** [review]
Thanks for making the changes @krisztina-zsihovszki!
LGTM+1

**Comment by mattyb149** [comment]
+1 LGTM, Merging to main and support/nifi-1.x

**Pull request closed without merging**